server/auth/github.js

```diff
@@ -0,0 +1,199 @@
+// @flow
+import passport from "@outlinewiki/koa-passport";
+import Router from "koa-router";
+import { Strategy as GithubStrategy } from "passport-github2";
+import accountProvisioner from "../commands/accountProvisioner";
+import env from "../env";
+import * as Github from "../github";
+import auth from "../middlewares/authentication";
+import passportMiddleware from "../middlewares/passport";
+import { Authentication, Collection, Integration, Team } from "../models";
+import { StateStore } from "../utils/passport";
+
+const router = new Router();
+const providerName = "github";
+const GITHUB_CLIENT_ID = process.env.GITHUB_KEY;
+const GITHUB_CLIENT_SECRET = process.env.GITHUB_SECRET;
+
+const scopes = ["repo", "read:org", "user"];
+
+export const config = {
+  name: "Github",
+  enabled: !!GITHUB_CLIENT_ID,
+};
+
+if (GITHUB_CLIENT_ID) {
+  const strategy = new GithubStrategy(
+    {
+      clientID: GITHUB_CLIENT_ID,
+      clientSecret: GITHUB_CLIENT_SECRET,
+      callbackURL: `${env.URL}/auth/github.callback`,
+      passReqToCallback: true,
+      store: new StateStore(),
+      scope: scopes,
+    },
+    async function (req, accessToken, refreshToken, profile, done) {
+      try {
+        const result = await accountProvisioner({
+          ip: req.ip,
+          team: {
+            name: profile.displayName,
+            avatarUrl: profile._json.avatar_url,
+          },
+          user: {
+            name: profile.username,
+            email: profile.emails[0].value,
```

Comment:
Is the first email always the default email? Or does the object include a key to identify which email in the array is default…

Reply:
I only have one email on the account, I can try to add more emails and see what I find.